src/utils/fetch.js

```diff
@@ -0,0 +1,59 @@
+/* globals fetch */
+import qs from 'querystring'
+
+const API_URL = `${process.env.GNOSISDB_URL}/api`
+
+export const requestFromRestAPI = async (endpoint, queryparams) => {
+  const url = `${API_URL}/${endpoint}?${qs.stringify(queryparams)}`
+  const response = await fetch(url)
+
+  if (response.status > 400) {
+    console.warn(`Couldn't fetch ${url}: ${response.status}: ${response.statusText}`)
+    throw new Error('GnosisDB: Couldn\'t fetch (invalid statuscode)')
+  }
+
+  let json
+  try {
+    json = response.json()
+  } catch (e) {
+    console.warn(`Couldn't format response to JSON: ${url}: ${response.body}`)
+    throw new Error('GnosisDB: Couldn\'t fetch (format error)')
+  }
+
+  return json
+}
+
+export const requestFromRestAPIPage = async (endpoint, queryparams = {}, page = 0, pageSize = 100) =>
+  requestFromRestAPI(endpoint, {
+    ...queryparams,
+    limit: pageSize,
+    size: pageSize,
+    offset: page * pageSize,
+  })
+
+export const requestFromRestAPIAllPages = async (endpoint, queryparams, pageSize, limit = 100000) => {
```

Comment:
This method should contain tests. At least: 
* Fetching one endpoint having 0 results
* Fetching one endpoint having 1 page 
* Fetching one endpoint having 2.7 pages (meaning the last one is not complete), fetching all pages.
* If you feel inspired and you want to add a stress test in order to know the limits, even better.